### PR TITLE
Add authentication stats

### PR DESCRIPTION
**Pull request by psiinon**
Makes it much easier to work out if authenticated scans are working ;)

### CODE REVIEW

**Comment by psiinon** [comment]
Ooops, I'll look at the failing tests..

**Comment by psiinon** [comment]
Hopefully thats ok now

**Comment by kingthorin** [comment]
Looks good.

File: src/org/zaproxy/zap/authentication/AuthenticationHelper.java [hunk 1]
```diff
@@ -31,8 +34,16 @@ public AuthenticationHelper(HttpSender httpSender, SessionManagementMethod sessi
 	private static final Logger log = Logger.getLogger(AuthenticationHelper.class);
 
 	private static final String HISTORY_TAG_AUTHENTICATION = "Authentication";
+	public static final String AUTH_SUCCESS_STATS = "stats.auth.success";
+	public static final String AUTH_FAILURE_STATS = "stats.auth.failure";
 
-	public static void notifyOutputAuthSuccessful() {
+	public static void notifyOutputAuthSuccessful(HttpMessage msg) {
```
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationHelper.java]: This breaks binary compatibility, since the next main release is expected to be binary compatible the "old method" should be deprecated instead.

File: src/org/zaproxy/zap/authentication/AuthenticationHelper.java [hunk 2]
```diff
@@ -40,7 +51,21 @@ public static void notifyOutputAuthSuccessful() {
...
+			// Ignore
+		}
+		// Let the user know it worked
+		if (View.isInitialised()) {
+			View.getSingleton().getOutputPanel()
+					.append(Constant.messages.getString("authentication.output.failure") + "\n");
+		}
+	}
+
+	public HttpState getCorrespondingHttpState(HttpMessage msg) {
```
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationHelper.java]: Was this change on purpose? The parameter is not being used.
Same comment as above regarding binary compatibility.

File: src/org/zaproxy/zap/authentication/AuthenticationHelper.java [hunk 3]
```diff
@@ -40,7 +51,21 @@ public static void notifyOutputAuthSuccessful() {
...
 
-	public HttpState getCorrespondingHttpState() {
+	public static void notifyOutputAuthFailure(HttpMessage msg) {
+		// Always record stats
+		try {
+			Stats.incCounter(SessionStructure.getHostName(msg), AUTH_FAILURE_STATS);
+		} catch (URIException e) {
+			// Ignore
+		}
+		// Let the user know it worked
```
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationHelper.java]: The code comment should be changed to "... it *not* worked" instead?

File: src/org/zaproxy/zap/authentication/AuthenticationHelper.java (hunk 1 above)
Comment (psiinon) [inline src/org/zaproxy/zap/authentication/AuthenticationHelper.java]: Good point, will fix.

File: src/org/zaproxy/zap/authentication/AuthenticationHelper.java (hunk 3 above)
Comment (psiinon) [inline src/org/zaproxy/zap/authentication/AuthenticationHelper.java]: Yeah ;)
